lms/djangoapps/certificates/api.py

```diff
@@ -610,14 +613,28 @@ def get_allowlist_entry(user, course_key):
     """
     Retrieves and returns an allowlist entry for a given learner and course-run.
     """
-    return CertificateWhitelist.objects.get(user=user, course_id=course_key)
+    log.info(f"Attempting to retrieve an allowlist entry for student {user.id} in course {course_key}.")
```

Comment:
Seemed like the logic of looking up the object for the learner is better suited here than in the instructor app. Followed a similar path to what the certificate lookup is doing.